src/solvers/flattening/boolbv.cpp

```diff
@@ -565,6 +565,11 @@ literalt boolbvt::convert_rest(const exprt &expr)
     else if(op.type().id() == ID_fixedbv)
       return const_literal(true);
   }
+  else if(expr.id() == ID_function_application)
+  {
+    functions.record(to_function_application_expr(expr));
+    return prop.new_variable();
+  }
```

Comment:
Does this also have implications outside SMT2, i.e., were uninterpreted functions previously broken in some way?

Reply:
Yeah; I agree with @tautschnig this seems like it should have a visible / testable effect on the usual back-end.

Reply:
> Does this also have implications outside SMT2, i.e., were uninterpreted functions previously broken in some way?

I've added a further test that uses the C front-end.